iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -545,6 +546,276 @@ TEST_F(PoshRuntime_test, GetMiddlewareSubscriberWithQueueFullPolicySetToBlockPub
                 Eq(iox::popo::QueueFullPolicy::BLOCK_PRODUCER));
 }
 
+TEST_F(PoshRuntime_test, GetMiddlewareClientWithDefaultArgsIsSuccessful)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2db35746-e402-443f-b374-3b6a239ab5fd");
+    const iox::capro::ServiceDescription sd{"911", "1", "20"};
+    iox::popo::ClientOptions defaultOptions;
+    iox::runtime::PortConfigInfo defaultPortConfigInfo;
+
+    auto clientPort = m_runtime->getMiddlewareClient(sd);
+
+    ASSERT_THAT(clientPort, Ne(nullptr));
+
+    EXPECT_EQ(clientPort->m_serviceDescription, sd);
+    EXPECT_EQ(clientPort->m_runtimeName, m_runtimeName);
+    EXPECT_EQ(clientPort->m_nodeName, defaultOptions.nodeName);
+    EXPECT_EQ(clientPort->m_connectRequested, defaultOptions.connectOnCreate);
+    EXPECT_EQ(clientPort->m_connectionState, iox::ConnectionState::WAIT_FOR_OFFER);
+    EXPECT_EQ(clientPort->m_chunkReceiverData.m_queue.capacity(), defaultOptions.responseQueueCapacity);
+    EXPECT_EQ(clientPort->m_chunkReceiverData.m_queueFullPolicy, defaultOptions.responseQueueFullPolicy);
+    EXPECT_EQ(clientPort->m_chunkReceiverData.m_memoryInfo.deviceId, defaultPortConfigInfo.memoryInfo.deviceId);
+    EXPECT_EQ(clientPort->m_chunkReceiverData.m_memoryInfo.memoryType, defaultPortConfigInfo.memoryInfo.memoryType);
+    EXPECT_EQ(clientPort->m_chunkSenderData.m_historyCapacity, iox::popo::ClientPortData::HISTORY_CAPACITY_ZERO);
+    EXPECT_EQ(clientPort->m_chunkSenderData.m_consumerTooSlowPolicy, defaultOptions.serverTooSlowPolicy);
+    EXPECT_EQ(clientPort->m_chunkSenderData.m_memoryInfo.deviceId, defaultPortConfigInfo.memoryInfo.deviceId);
+    EXPECT_EQ(clientPort->m_chunkSenderData.m_memoryInfo.memoryType, defaultPortConfigInfo.memoryInfo.memoryType);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareClientWithCustomClientOptionsIsSuccessful)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "f61a81f4-f610-4e61-853b-ac114d9a801c");
+    const iox::capro::ServiceDescription sd{"911", "1", "20"};
```

Comment:
You could think about having those different to the first test case, if you want to be über-sure.